### PR TITLE
Deprecate gather, gatherN, gatherUnordered in favor of parSequence, parSequenceN, parSequenceUnordered

### PR DESCRIPTION
**Pull request by jozic**
addresses #1130 

- [x] gatherX to parSequenceX
- [x] wanderX to parTraverseX
- [ ] update https://github.com/monix/monix.io

### CODE REVIEW

**Comment by jozic** [comment]
@Avasil @oleg-py @alexandru 
please check this out, if this looks good, i can continue with the rest

File: monix-eval/shared/src/main/scala/monix/eval/Task.scala [hunk 1]
```diff
@@ -3517,36 +3517,46 @@ object Task extends TaskInstancesLevel1 {
     implicit bf: BuildFrom[M[A], B, M[B]]): Task[M[B]] =
     TaskSequence.traverse(in, f)(bf)
 
+  /** Alias for [[parSequence]] */
+  @deprecated("Use parSequence", "3.2.0")
+  def gather[A, M[X] <: Iterable[X]](in: M[Task[A]])(implicit bf: BuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] =
```
Comment (Avasil) [inline monix-eval/shared/src/main/scala/monix/eval/Task.scala]: We should put deprecated methods to `TaskDeprecated.Companion` to clean normal API

There is also slightly different convention for description with
```
/**
   * DEPRECATED — please use XXX
   */
```

**Comment by jozic** [comment]
@Avasil thanks for feedback!
 I've moved deprecated methods to `TaskDeprecated.Companion` and also added renamed classes to the mima filter
I've also tried to revert changes to benchmarks, but current version wasn't compilable due to deprecation errors, so I've disabled `benchmarksPrev`
CI seems happy now

Please check it out again

**Comment by Avasil** [comment]
@jozic Looks good, we can add the rest :+1:

I think we can wait with `monix.io` docs update until `3.2.0` is released

**Comment by jozic** [comment]
@Avasil I've added second commit with changes for `wander` and friends
lmk if anything need to be changed

**Pull request merged**